Components/map

gmafrac opened the pull request:
<!--PS: 🇧🇷/🇬🇧 It's ok to write your PR in portuguese 😃-->

## What type of PR is this? (check all applicable)
- [x] ✨ Feature
- [ ] 🐞 Bug Correction
- [ ] 📝 Documentation Update
- [ ] 🚩 Other

# Changes:
As mudanças requisitadas por esse PR são relacionadas a interface desenvolvida para o componente Mapa. Nela, é possível visualizar o mapa com as respectivas posições da sonda e do usuário. O intuito dessa aplicação é desenvolver futuramente, por meio de uma API já determinada, a criação de rotas envolvendo esse percurso.

![image](https://user-images.githubusercontent.com/81496731/152370529-a06ca46b-58b7-4e6e-94af-96e11ce557de.png) 

Ademais, esta interface é responsável por expor alguma das variáveis obtidas da alcântara. Ainda não foi desenvolvido a passagem dos valores do Firebase para ela, mas futuramente essa conexão será desenvolvida. Nesta interface é possível com o toque, puxar, e se orientar nos demais containers que envolvem as informações recebidas.



![image](https://user-images.githubusercontent.com/81496731/152371386-73af1506-68e0-47c9-93f8-894d248374ff.png)



Outra parte desenvolvida, é o botão de tema que por meio dele é possível modificar a temática do mapa. Ao começo estava em dúvida de qual tema utilizar entre ambos e, com a opinião do grupo, desenvolvi essa aplicação para haver a mudança do tema mais azulado e "noturno" para uma temática de satélite, a qual combina com a ideia do projeto, de desenvolver uma aplicação que contribua para o projeto de uma sonda aeroespacial.



![image](https://user-images.githubusercontent.com/81496731/152370455-aafa4292-ee76-4428-9c50-301699edd072.png)



![image](https://user-images.githubusercontent.com/81496731/152371153-1dfc3c11-7ac0-4297-a043-772a1f67c3ee.png)



Por fim, esta tela é uma ponte de conexão entre outras demais telas. Esta relação se dá por meio da tela do drawer, que é chamada tanto quando o botão a esquerda da tela é puxado quando clicado. Por ela, se relaciona tanto as informações do usuário cadastrado, que possui seus dados e uma imagem para identifica-lo, quanto botões que >futuramente< serão a ponte para mudança de tela.



![image](https://user-images.githubusercontent.com/81496731/152371442-8c982fae-5c82-4f02-bc5b-77705a65fe25.png)



![image](https://user-images.githubusercontent.com/81496731/152371500-65bb2624-6669-4c9e-95bc-b567911e1ba5.png)



Assim, peço uma revisão de vocês. Ainda falta algumas implementações que futuramente irei realizar como a mudança dos ícones do mapa por bitmaps de uma sonda e de um usuário para, assim, identificar as posições de forma mais nítida...
Outra coisa... eu não sabia como diminuir o tamanho das imagens então acabou ficando assim mesmo :(

 # Testing
<!--(if aplicable) Tell us how to replicate the expected behaviour.-->

gmafrac committed:
testing a google maps demo in main.dart

gmafrac committed:
create map.dart and adding google map and its style

gmafrac committed:
map.dart: changes some configurations

gmafrac committed:
map.dart: adding informations containers and a half-circle

gmafrac committed:
For rebase with develop

gmafrac committed:
Merge map with develop

gmafrac committed:
Map: Adding theme button, modifying drawer and fixing info containers

gmafrac committed:
Map: Adding theme button, modifying drawer and fixing info containers

gmafrac commented:
Um adendo que havia esquecido de comentar durante o PR: o código em si ainda não está responsivo pois não consegui visualizar ele em diferentes resoluções. O chrome não está sendo capaz de gerar a interface e por isso eu precisava de alguma sugestão também para esse problema, pois os devices do android studio são muito pesados para conseguir testar esse aspecto.

maxsuelfalmeida commented on android/app/src/main/AndroidManifest.xml:
```diff
@@ -5,6 +5,8 @@
         android:label="zenith_monitor"
         android:icon="@mipmap/ic_launcher">
+        <meta-data android:name="com.google.android.geo.API_KEY"
+               android:value= API KEY/>
```
Acredito que o erro é causado por esta linha.
```suggestion
               android:value= "API KEY"/>
```

maxsuelfalmeida reviewed (commented):
Tentei rodar e ocorreu o erro mostrado na imagem abaixo:

![Screenshot from 2022-02-03 12-52-25](https://user-images.githubusercontent.com/61526540/152387139-7fc053e1-c17b-44b8-b65d-34e126392fd0.png)

VitorFCM commented:
Tive o mesmo problema que o @maxsuelfalmeida, vou dar uma olhada no que pode ser.

VitorFCM commented on android/app/src/main/AndroidManifest.xml:
```diff
@@ -5,6 +5,8 @@
    <application
         android:label="zenith_monitor"
         android:icon="@mipmap/ic_launcher">
+        <meta-data android:name="com.google.android.geo.API_KEY"
```
@maxsuelfalmeida, o problema que tivemos tem a ver com esta linha, não temos a chave da api para carregar o mapa. @g-mafra, em alguma reunião seria interessante você mostrar como utilizar a chave.

gmafrac committed:
Fixing details in AndroidManifest

gmafrac committed:
Refactoring map.dart: Fixing responsiveness and orientation errors.

VitorFCM committed:
Adding a DraggableScrollableSheet over real time variables

VitorFCM committed:
Just changing a container size to make it more responsable

VitorFCM committed:
Removing unnecessary print screen

VitorFCM committed:
This image is no longer necessary

VitorFCM committed:
Changing the code of information cards

Instead of using functions that return widgets, a Stateless class was created to return information cards

VitorFCM committed:
Just changing the name of the map widget

VitorFCM committed:
Moving the map widget from widgets/map/ directory to widgets/

VitorFCM committed:
Refactoring the map_theme_button.dart code

VitorFCM committed:
Refactoring the navigation_drawer.dart code

VitorFCM committed:
Returning class to the way it was in develop branch

VitorFCM committed:
Adapting components to remove timestamp variable from MissionVariablesList

VitorFCM committed:
Reverting changes to branch develop

The code that had been implemented was used only for the development of the map screen and for the deletion of the timestamp variable, which was no longer necessary

VitorFCM committed:
Removing files from widgets/map/ and putting them in widgets/

VitorFCM committed:
Adding quotes in fake api key for application not to break

VitorFCM committed:
Merge branch 'develop' into components/map

VitorFCM committed:
Just solving error generated during merge with develop
